CppCoreGuidelines.md

```diff
@@ -18747,6 +18747,7 @@ Source file rule summary:
 * [SF.20: Use `namespace`s to express logical structure](#Rs-namespace)
 * [SF.21: Don't use an unnamed (anonymous) namespace in a header](#Rs-unnamed)
 * [SF.22: Use an unnamed (anonymous) namespace for all internal/non-exported entities](#Rs-unnamed2)
+* [SF.23: Keep forward declarations independent](#Rs-forward)
```

Comment:
The title of this guideline is extremely general, but the text of the guideline is very specific to nested namespaces. From the title, I expected to see some discussion of pathological forward declarations like

    int f(int x, int y=0);   // if you omit this declaration...
    int f(int x=0, int y);   // ...then this declaration becomes ill-formed

or

    struct S;   // if you omit this declaration...
    namespace N { int f(struct S); }  // ...then this declaration changes meaning from ::S to N::S

Reply:
I agree, the rule sounds more exciting than it is. Can you suggest a more appropriate name? I guess I could say something like "Start each forward declaration from scratch" or something.

On that note it's probably not clear but most of my use of forward declarations is with classes so I'm really only talking about class forward declarations here. I haven't considered whether it applies to function forward declarations. My instinct is that it does except that method declarations tend to be a lot longer so where for a class it would be one line, for a method it would be several which loses a lot of the benefit.